asset-transfer-private-data/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,412 @@
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'keystore'
+);
+
+// Path to org2 user certificate.
+const certPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'users',
+    'User1@org2.example.com',
+    'msp',
+    'signcerts',
+    'cert.pem'
+);
+
+// Path to org2 peer tls certificate.
+const tlsCertPathOrg2 = path.resolve(
+    cryptoPathOrg2,
+    'peers',
+    'peer0.org2.example.com',
+    'tls',
+    'ca.crt'
+);
+
+// Gateway peer endpoint.
+const peerEndpointOrg1 = 'localhost:7051';
+const peerEndpointOrg2 = 'localhost:9051';
+
+// Gateway peer container name.
+const peerContainerOrg1 = 'peer0.org1.example.com';
+const peerContainerOrg2 = 'peer0.org2.example.com';
+
+const utf8Decoder = new TextDecoder();
+
+//Collection Names
+const org1PrivateCollectionName = 'Org1MSPPrivateCollection';
+const org2PrivateCollectionName = 'Org2MSPPrivateCollection';
+
+const RED = '\x1b[31m\n';
+const RESET = '\x1b[0m';
+
+//Use a random key so that we can run multiple times
+const randomNumber = Math.floor(Math.random() * 1000) + 1;
+const assetID1 = `asset${randomNumber}`;
+const assetID2 = `asset${randomNumber + 1}`;
+
+async function main(): Promise<void> {
+    // The gRPC client connection from org1 should be shared by all Gateway connections to this endpoint.
+    const clientOrg1 = await newGrpcConnection(
+        tlsCertPathOrg1,
+        peerEndpointOrg1,
+        peerContainerOrg1
+    );
+
+    const gatewayOrg1 = connect({
+        client: clientOrg1,
+        identity: await newIdentity(certPathOrg1, mspIdOrg1),
+        signer: await newSigner(keyDirectoryPathOrg1),
+    });
+
+    // The gRPC client connection from org2 should be shared by all Gateway connections to this endpoint.
+    const clientOrg2 = await newGrpcConnection(
+        tlsCertPathOrg2,
+        peerEndpointOrg2,
+        peerContainerOrg2
+    );
+
+    const gatewayOrg2 = connect({
+        client: clientOrg2,
+        identity: await newIdentity(certPathOrg2, mspIdOrg2),
+        signer: await newSigner(keyDirectoryPathOrg2),
+    });
+
+    try {
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg1 = gatewayOrg1.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg1 = networkOrg1.getContract(chaincodeName);
+
+        // Get a network instance representing the channel where the smart contract is deployed.
+        const networkOrg2 = gatewayOrg2.getNetwork(channelName);
+
+        // Get the smart contract from the network.
+        const contractOrg2 = networkOrg2.getContract(chaincodeName);
+
+        console.log('\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~');
+
+        // Create new assets on the ledger.
+        await createAsset(contractOrg1);
+
+        //Read  asset from the org1's private data collection.
+        await readAssetPrivateDetails(contractOrg1,org1PrivateCollectionName,assetID1);
```

Comment:
This function takes the asset ID as a parameter, whereas all the others use the asset ID const variable directly. It would be good to be consistent, whichever way 'round you prefer.

Reply:
It's not the collection name I'm questioning; it's the asset ID parameter. You only ever pass `assetID1` as the value so you could avoid passing that in as an argument and just have the **readAssetPrivateDetails()** function hard-coded to use `assetID1`, just like all the other functions